### PR TITLE
Adding TrendForecaster in forecasting/trend.py

### PR DESCRIPTION
**Pull request by tensorflow-as-tf**
#961 

It makes a new class named TrendForecaster that doesn't have the polynomial features at, while leaving the PolynomialTrendForecaster unchanged.

The contribution doesn't introduce a new dependency.

Feedback from the reviewer would be very much appreciated. They should look at code quality as well and comment on the readability of line 70 that creates a new variable ("regressor").

##### For new estimators
- [ ] I've added the estimator to the online documentation.

### CODE REVIEW

File: sktime/forecasting/trend.py [hunk 1]
```diff
@@ -16,9 +16,103 @@
...
+            The forecast horizon with the steps ahead to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        # for default regressor, set fit_intercept=False
+        regressor = self.regressor or LinearRegression(fit_intercept=False)
```
Comment (fkiraly) [inline sktime/forecasting/trend.py:69]: I'd make `fit_intercept=True` here.

File: sktime/forecasting/trend.py [hunk 2]
```diff
@@ -16,9 +16,103 @@
...
+
+        # make pipeline with regressor
+        self.regressor_ = make_pipeline(regressor)
```
Comment (fkiraly) [inline sktime/forecasting/trend.py:72]: why do you need that? There's just one component

File: sktime/forecasting/trend.py [hunk 3]
```diff
@@ -16,9 +16,103 @@
...
-    Forecast time series data with a polynomial trend.
+
+    _tags = {
+        "univariate-only": True,
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+    }
+
+    def __init__(self, regressor=None):
+        self.regressor = regressor
```
Comment (fkiraly) [inline sktime/forecasting/trend.py:48]: I think this should be a `clone`, i.e., use `sklearn` `clone` to make a copy to `regressor_`. No need to have two variables, `regressor` and `regressor_`

File: sktime/forecasting/trend.py [hunk 4]
```diff
@@ -16,9 +16,103 @@
...
+        """
+        # for default regressor, set fit_intercept=False
+        regressor = self.regressor or LinearRegression(fit_intercept=False)
+
+        # make pipeline with regressor
+        self.regressor_ = make_pipeline(regressor)
+
+        # transform data
+        n_timepoints = _get_duration(self._y.index, coerce_to_int=True) + 1
+        X = np.arange(n_timepoints).reshape(-1, 1)
```
Comment (fkiraly) [inline sktime/forecasting/trend.py:76]: here, I'd take into account the case where time points are not equally spaced. The trend strategy still works in that case, you just have to be careful with reformatting.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Good contribution, almost ready to go!

I made some comments above for change:

* there should just be a single `regressor_` attribute
* `regressor` should be cloned before assigning in `__init__`
* `make_pipeline` seems unnecessary
* the time points should be converted so the trend strategy still works for unequally spaced time index (if you know how to do it)

File: sktime/forecasting/trend.py (hunk 3 above)
Comment (mloning) [reply sktime/forecasting/trend.py:48]: Just to clarify, cloning happens in `_fit`, the `__init__` method just remembers the input.

File: sktime/forecasting/trend.py [hunk 5]
```diff
@@ -1,24 +1,117 @@
 #!/usr/bin/env python3 -u
 # -*- coding: utf-8 -*-
-# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""copyright: sktime developers, BSD-3-Clause License (see LICENSE file)."""
 
-__author__ = ["Markus Löning"]
-__all__ = ["PolynomialTrendForecaster"]
+__author__ = ["Anthony Jancso", "mloning"]
```
Comment (fkiraly) [inline sktime/forecasting/trend.py:5]: this should be your GitHub ID

**Review by fkiraly — APPROVED** [review]
Looks good, except that the author field should contain GitHub name.

**Comment by mloning** [comment]
Great work @tensorflow-as-tf! Could we also add this to the online docs? It should go into `docs/api_reference/forecasting.rst`.

**Comment by tensorflow-as-tf** [comment]
I added the function to the online docs in `docs/api_reference/forecasting.rst` in PR #1292

**Pull request merged**